src/java.base/share/classes/sun/security/ssl/CertificateMessage.java

```diff
@@ -1077,23 +1079,18 @@ private static SSLPossession choosePossession(
                     checkedKeyTypes.add(ss.keyAlgorithm);
                     continue;
                 }
-
-                SSLPossession pos = ka.createPossession(hc);
-                if (pos == null) {
-                    if (SSLLogger.isOn && SSLLogger.isOn("ssl,handshake")) {
-                        SSLLogger.warning(
-                            "Unavailable authentication scheme: " + ss.name);
-                    }
-                    continue;
-                }
-
-                return pos;
+                allAuths.add(ss.keyAlgorithm);
```

Comment:
As the use of HashSet, duplicated key algorithms could be filter out here.  It's a good point.  But the checking of the previous steps are still duplicated.  We may be able to have an improvement to avoid the duplicated checking.  See the comment at the declaration of allAuths.